### ISSUE TITLE
Promotion of Users with Unverified Emails

### ISSUE THREAD

**Issue by jsharkey13** [ucam-cl-dtg/isaac-app#775]
If a user has not verified their email address, we have no idea who they are. Currently you can promote users from the admin pages without any warnings about non-verified emails. 

The following code would prevent promotion of users who are not verified:
```java
// If first setting of user's role, or a promotion: they must be verified, else abort
if ((user.getRole() == null || (user.getRole().ordinal() <= requestedRole.ordinal())) &&
        !user.getEmailVerificationStatus().equals(EmailVerificationStatus.VERIFIED)) {
    return new SegueErrorResponse(Status.FORBIDDEN,
            "Aborted - cannot promote one or more users "
                    + "who have not verified their email addresses!").toResponse();
}
```
mimicking [that used to check role escalation/demotion attacks](https://github.com/ucam-cl-dtg/isaac-api/blob/master/src/main/java/uk/ac/cam/cl/dtg/segue/api/AdminFacade.java#L343-LL348).

This requires some discussion, since it might be annoying for in-person events to have everyone verify their email address before promoting them; when you can check that they are physically who they say they are even without a verified email. But it would prevent accidental promotion of people who should first have verified their email address. @allydavies @hjp22 - any thoughts?

Perhaps the most useful course of action is a confirmation message *per user-to-be-promoted* if they have not verified their email. This might irritate, but wouldn't stop it working. This check could be added to `modifySelectedUsersRole()` in [`AdminUserManagerController.js`](https://github.com/ucam-cl-dtg/isaac-app/blob/master/app/js/app/controllers/AdminUserManagerController.js#L140-L153).

**Comment by allydavies** [comment]
I agree it would be better to prevent easy promotion of those who haven't verified their email address.

I think it's a good thing to expect teachers to verify their email (and we know they do sometimes enter their emails incorrectly).

An extra thought - what would happen to a teacher who's email bounced? Would it be good to get them  to lose teacher status (until they edit their email and verify new email)?

I think the in most cases we could ask that all users verify first - we do ask them to enter school first.

I would be happy with either behaviour - i.e. CANNOT promote unverified OR warning before able to promote unverified.
